estuary_updater/handlers/freshmaker.py

```diff
@@ -79,13 +80,57 @@ def build_state_handler(self, msg):
         """
         build_info = msg['body']['msg']
         event_id = msg['body']['msg']['event_id']
+        freshmaker_build = self.create_or_update_freshmaker_build(build_info, event_id)
         build = self.create_or_update_build(build_info, event_id)
-        if build:
-            event = FreshmakerEvent.nodes.get_or_none(id_=str(event_id))
-            if event:
-                event.triggered_container_builds.connect(build)
-            else:
-                log.warn('The Freshmaker event {0} does not exist in Neo4j'.format(event_id))
+        event = FreshmakerEvent.nodes.get_or_none(id_=str(event_id))
+        if event:
+            if build:
+                event.successful_koji_builds.connect(build)
+            if freshmaker_build:
+                freshmaker_build.event.connect(event)
+                event.requested_builds.connect(freshmaker_build)
+        else:
+            log.warn('The Freshmaker event {0} does not exist in Neo4j'.format(event_id))
+
+    def create_or_update_freshmaker_build(self, build, event_id):
+        """
+        Create or update a FreshmakerBuild.
+
+        :param dict build: the build represented in Freshmaker being created or updated
+        :param int event_id: the id of the Freshmaker event
+        :return: the created/updated FreshmakerBuild or None if it cannot be created
+        :rtype: FreshmakerBuild or None
+        """
+        # build_id in Freshmaker is actually the task_id
+        if not build['build_id']:
```

Comment:
This `if` and `elif` duplicate the ones in `create_or_update_build`. Could you please either move these conditionals to `build_state_handler` or to a separate private method?